File: injector/src/test/java/org/codejargon/feather/NamedDependencyTest.java
package org.codejargon.feather;

import org.junit.Test;

import javax.inject.Named;

import static org.junit.Assert.assertEquals;

public class NamedDependencyTest {
    @Test
    public void namedInstanceWithModule() {
        Injector injector = Injector.with(new HelloWorldModule());
        assertEquals("Hello!", injector.instance(Key.of(String.class, "hello")));
        assertEquals("Hi!", injector.instance(Key.of(String.class, "hi")));
    }

    public static class HelloWorldModule {
        @Provides
        @Named("hello")
        String hello() {
            return "Hello!";
        }

        @Provides
        @Named("hi")
        String hi() {
            return "Hi!";
        }
    }

}

File: injector/src/test/kotlin/net/mready/inject/NamedDependencyTest.kt
package net.mready.inject

import org.junit.Assert.assertEquals
import org.junit.Test
import javax.inject.Named

class NamedDependencyTest {
    @Test
    fun namedInstanceWithModule() {
        val injector = Injector.with(HelloWorldModule())
        assertEquals("Hello!", injector.instance(Key.of(String::class.java, "hello")))
        assertEquals("Hi!", injector.instance(Key.of(String::class.java, "hi")))
    }

    class HelloWorldModule {
        @Provides
        @Named("hello") fun hello(): String {
            return "Hello!"
        }

        @Provides
        @Named("hi") fun hi(): String {
            return "Hi!"
        }
    }

}
